openshift/hac-dev
# use cdq and cluster defaults to determine in resource limit validation.

## Fixes 

https://issues.redhat.com/browse/RHTAPBUGS-493

## Description


When cdq returned resource limits, our resource limit validation was preventing user to submit the form until the decrease the values to match our hardcoded defaults. (10m, 256Mi).

This PR covers the following

1.  Fetch limitRange CR to determine the min/max values of the cluster.
2.  Use CDQ defaults if available or fallback to cluster default min limit.
3.  Use cluster default's max limit in the validation utils, so user can increase the value upto this limit.
4.  Added unit conversion utils to convert the values based on selected unit and display right values in the error message. 



## Type of change




## Screen shots / Gifs for design review 


Before:

**Form button disabled:**
<img width="1497" alt="Screenshot 2023-07-28 at 6 15 11 PM" src="https://github.com/openshift/hac-dev/assets/9964343/60115a7b-4eb5-4aec-8295-d103d7932108">


After:


<img width="1501" alt="Screenshot 2023-07-28 at 6 14 30 PM" src="https://github.com/openshift/hac-dev/assets/9964343/ed943ea7-817a-43ff-8daf-49099487d59c">

## How to test or reproduce?







1. Use this repository https://github.com/openshift-cnv/cnv-fbc
2. Click on Import, in the review step the create application button should be enabled.


## Browser conformance:
